t/integration/test_canvas.py

```diff
@@ -1274,6 +1275,19 @@ def test_nested_chord_group_chain_group_tail(self, manager):
         res = sig.delay()
         assert res.get(timeout=TIMEOUT) == [[42, 42]]
 
+    @pytest.mark.xfail(TEST_BACKEND.startswith('redis://'), reason="Issue #6437")
+    def test_error_propagates_from_chord(self, manager):
+        try:
+            manager.app.backend.ensure_chords_allowed()
+        except NotImplementedError as e:
+            raise pytest.skip(e.args[0])
+
+        sig = add.s(1, 1) | fail.s() | group(add.s(1), add.s(1))
```

Comment:
It might be better to construct the chord signature manually for test cases rather than relying on the chain->group promotion behaviour.

Reply:
But perhaps that's what makes the bug since `test_chord_on_error()` passes :o worth making the change if you have the time to see if we can narrow down what's breaking

Reply:
You're right, it times out when the failing task is in the middle of the chain. I'm sorry for the confusion @thedrow